src/Adapter/MSTestAdapter.PlatformServices/Services/TestSourceHost.cs

```diff
@@ -125,10 +119,17 @@ public void SetupHost()
         {
             assemblyResolver.Dispose();
         }
+#elif NETFRAMEWORK
+        List<string> resolutionPaths = GetResolutionPaths(_sourceFileName, VSInstallationUtilities.IsCurrentProcessRunningInPortableMode());
 
-#endif
+        if (EqtTrace.IsInfoEnabled)
+        {
+            EqtTrace.Info("DesktopTestSourceHost.SetupHost(): Creating assembly resolver with resolution paths {0}.", string.Join(",", resolutionPaths));
+        }
+
+        _parentDomainAssemblyResolver = new AssemblyResolver(resolutionPaths);
```

Comment:
Removing the `Dispose` alone or the assignment alone isn't enough. It's really not clear why things need to be done this way.

While writing that, I realize I need to add a comment in code.